Error when importing data

Using data files as directed in the readme the files report as not being found. Specifically says `Cannot find module 'C:somepathsomeprojectdataindex-meta.js'` note when this happens the error message that shows is completely devoid of forward slashes which could be the cause of the issue. 

To reproduce the error simply try to use <% const meta= require('../data/index-meta.js') %> or any path with data. 

The data file is structured as shown in the readme literally exactly 

```
module.exports = {
  title: 'Webpack Starter App',
  author: 'John Doe',
  keywords: ['lorem', 'ipsum', 'dolor', 'sit', 'amet'],
  description: 'Lorem ipsum dolor sit, amet consectetur adipisicing elit.',
  customFunction: function () {
    // ...
  },
}

```

--- a/package.json
+++ b/package.json
@@ -24,12 +24,13 @@
     "lib/"
   ],
   "scripts": {
-    "build": "tsc src/*.ts --outDir lib -d",
+    "prebuild": "rimraf lib",
+    "build": "tsc --project tsconfig.build.json",
     "lint": "eslint . --ext .ts --cache",
     "lint:fix": "eslint . --ext .ts --cache --fix",
     "lint-staged": "lint-staged",
     "prepare": "husky install",
-    "test:webpack-build": "NODE_ENV=test webpack --config __tests__/webpack.config.js",
+    "test:webpack-build": "cross-env NODE_ENV=test webpack --config __tests__/webpack.config.js",
     "test": "yarn build && yarn test:webpack-build && jest",
     "tsc": "tsc"
   },
@@ -51,6 +52,7 @@
     "@types/node": "16.11.7",
     "@typescript-eslint/eslint-plugin": "5.27.0",
     "@typescript-eslint/parser": "5.27.0",
+    "cross-env": "7.0.3",
     "eslint": "8.16.0",
     "eslint-config-prettier": "8.5.0",
     "eslint-plugin-import": "2.26.0",
@@ -60,6 +62,7 @@
     "jest": "28.1.0",
     "lint-staged": "12.4.3",
     "prettier": "2.6.2",
+    "rimraf": "3.0.2",
     "ts-jest": "28.0.3",
     "typescript": "4.7.2",
     "webpack": "5.72.1",

--- a/src/index.ts
+++ b/src/index.ts
@@ -27,9 +27,7 @@ const resolveRequirePaths = async (context: EjsLoaderContext, content: string) =
 
     if (requestSource !== null) {
       const result = await context.getResolve()(context.context, requestSource)
-      // eslint-disable-next-line no-console
-      console.log(result)
-      resultContent = resultContent.replace(matches[1], `require('${result}')`)
+      resultContent = resultContent.replace(matches[1], `require('${resolve(result)}')`)
     }
 
     matches = requirePattern.exec(content)

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,5 +1,5 @@
 import { readFileSync } from 'fs'
-import { resolve } from 'path'
+import { resolve, sep } from 'path'
 
 import { compile, Options, Data } from 'ejs'
 import { LoaderContext } from 'webpack'
@@ -26,8 +26,11 @@ const resolveRequirePaths = async (context: EjsLoaderContext, content: string) =
     const requestSource = matchFilename !== null ? matchFilename[0].replace(/['"`]/g, '') : null
 
     if (requestSource !== null) {
-      const result = await context.getResolve()(context.context, requestSource)
-      resultContent = resultContent.replace(matches[1], `require('${resolve(result)}')`)
+      let result = await context.getResolve()(context.context, requestSource)
+      if (sep === '\\') {
+        result = result.replace(/\\/g, '\\\\')
+      }
+      resultContent = resultContent.replace(matches[1], `require('${result}')`)
     }
 
     matches = requirePattern.exec(content)

--- a/src/index.ts
+++ b/src/index.ts
@@ -27,6 +27,8 @@ const resolveRequirePaths = async (context: EjsLoaderContext, content: string) =
 
     if (requestSource !== null) {
       const result = await context.getResolve()(context.context, requestSource)
+      // eslint-disable-next-line no-console
+      console.log(result)
       resultContent = resultContent.replace(matches[1], `require('${result}')`)
     }
 

--- a/tsconfig.build.json
+++ b/tsconfig.build.json
@@ -1,0 +1,7 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "lib"
+  },
+  "include": ["src/**/*.ts"]
+}

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -3,7 +3,6 @@
     "target": "ES2020",
     "module": "commonjs",
     "sourceMap": true,
-    "outDir": "./lib",
     "strict": true,
     "esModuleInterop": true,
     "skipLibCheck": true,

